proposals/0454-task-naming-api.md

```diff
@@ -0,0 +1,231 @@
+
+## Proposed solution
+
+In order to allow developers to provide helpful names for Swift Tasks, the Swift Task creation APIs should be modified to *optionally* allow developers to provide a name for that task.
+
+Consider the example:
+```swift
+let getUsers = Task {
+	await users.get(accountID))
+}
+```
+
+In order to ease debugging, a developer could create this unstructured task by passing in a name instead:
+
+```
+let getUsers = Task(name: "Get Users") {
+	await users.get(accountID)
+}
+```
+
+Or, if a developer has a lot of similar Tasks, they can provide more contextual information using string interpolation.
+
+```
+let getUsers = Task("Get Users for \(accountID)") {
+	await users.get(accountID)
+}
+```
+
+By introducing this in a structured, coherent way in Swift itself, rather than developers each inventing their own task-local with a name, runtime inspection tools and debuggers can become aware of task names and show you exactly which accountID was causing the crash or a profiling tool could tell you which accountID request was slow to load.
+
+## Detailed design
+
+
+Naming tasks is only allowed during their creation, and modifying names is not allowed. Names are arbitrary user-defined strings, which may be computed at runtime because they often contain identifying information such as the request ID or similar runtime information.
+
+In order to allow naming tasks, the following APIs will be provided on `Task`:
+
+```swift
+  init(
+     name: String? = nil,
+     priority: TaskPriority? = nil,
+     operation: sending @escaping @isolated(any) () async -> Success
+     
+  static func detached(
+     name: String? = nil,
+     priority: TaskPriority? = nil,
+     operation: sending @escaping @isolated(any) () async -> Success
+```
+
+In addition to these APIs to name unstructured Tasks, the following API will be added to name Tasks created as a part of a task group task group:
+
+```swift
+mutating func addTask(
+    name: String? = nil,
+    executorPreference taskExecutor: (any TaskExecutor)? = nil,
+    priority: TaskPriority? = nil,
+    operation: sending @escaping @isolated(any) () async -> ChildTaskResult
+  )
+  
+  mutating func addTaskUnlessCancelled(
+    name: String? = nil,
+    executorPreference taskExecutor: (any TaskExecutor)? = nil,
+    priority: TaskPriority? = nil,
+    operation: sending @escaping @isolated(any) () async -> ChildTaskResult
+  )
+
+```
+
+These APIs would be added to all kinds of task groups, including throwing, discarding ones. With the signature being appropriately matching the existing addTask signatures of those groups
+
+In addition to that, it will be possible to read a name off a task, similar to how the current task's priority is possible to be read:
+
+```swift
+extension Task {
+  var currentName: String? { get } 
+}
+```
+
+### Actor Identity
```

Comment:
Maybe let's move it into alternatives considered, since we're not proposing any actual change here.